src/cli.rs

```diff
@@ -949,6 +957,51 @@ impl Pretty {
     }
 }
 
+#[derive(Debug)]
+pub struct FormatOptions {
+    pub json_indent: usize,
+}
+
+impl Default for FormatOptions {
+    fn default() -> Self {
+        Self {
+            json_indent: 4,
+        }
+    }
+}
+
+impl FromStr for FormatOptions {
+    type Err = anyhow::Error;
+    fn from_str(options: &str) -> anyhow::Result<FormatOptions> {
+        let mut format_options = FormatOptions::default();
+
+        // generate a map of the specified options
+        for argument in options.split(',') {
+            let argument_parts = argument.split(':').collect::<Vec<&str>>();
+            // no proper argument provided, thus skipping
+            if argument_parts.len() < 2 {
+                return Err(anyhow!(
+                    "Format options consist of a key and a value, separated by a \":\"."
+                ));
+            }
+
+            let key = argument_parts[0];
+            let value = argument_parts[1];
+
+            // save the specified options to the `FormatOptions` struct
+            if key == "json.indent" {
+                if let Ok(json_indent) = value.parse::<usize>() {
+                    format_options.json_indent = json_indent;
+                }
+            } else {
+                return Err(anyhow!("Format option \"{}\" not supported", key));
+            }
+        }
```

Comment:
We can make this more strict by disallowing options with multiple colons e.g `json.indent:8:8` or invalid indent number e.g `json.indent:-2`
 
```suggestion
        for argument in options.split(',') {
            let argument_parts: Vec<&str> = argument.splitn(2, ':').collect();
            match argument_parts[..] {
                ["json.indent", value] => {
                    format_options.json_indent =
                        value.parse().map_err(|_| anyhow!("Invalid indent value"))?;
                }
                _ => {
                    return Err(anyhow!(
                        "Format options consist of a key and a value, separated by a \":\"."
                    ));
                }
            }
        }
```

Reply:
Really neat, I wasn't aware of that handy match expression using `[..]`